### PR TITLE
Feature/add list of asset values to genereted file

### PR DESCRIPTION
**Pull request by zemcov-surf**
### Description of the Change
Add static const List<String> values parameter to the genereted files.
```dart
class Images {
  Images._();

  static const List<String> values = [search, home, backArrow];

  static const String search = 'assets/search.png';
  static const String home = 'assets/home.png';
  static const String backArrow = 'assets/back_arrow.png';
}
```

### Why Should This Be In Core?

User can use variable values to iterate all assets like this:
```dart
for (final icon in Images.values) {
  /*do something cool*/
}
```

### Benefits

For exapmle, user can to generate an image with all the possible icons in the application like this:
```dart
void main() {
  testGoldens('Icons golden test', (tester) async {
    final builder = GoldenBuilder.column();

    for (final imagePath in Images.values) {
      builder.addScenario(
        iconPath,
        Image.asset(
          iconPath,
          height: 24,
          width: 24,
        ),
      );
    }

    await tester.pumpWidgetBuilder(
      builder.build(),
      surfaceSize: const Size(420, 3650),
      wrapper: materialAppWrapper(theme: AppTheme.defaultTheme),
    );

    await screenMatchesGolden(tester, 'icons');
  });
}
```

### Possible Drawbacks

In the generated asset file we have now a variable `static const List<String> values` that contains list of the all genereted assets.

### Verification Process

I verified new functionality with data_class_generator_test.dart. I used debug points and also added one test case to the test class.

### Applicable Issues

#38

### CODE REVIEW

File: lib/src/data/json_config.dart [hunk 1]
```diff
@@ -20,6 +20,7 @@ const String JSON_CONFIGS = '''{
   "no_comments":true,
   "export":true,
   "use_part_of":true,
+  "use_references_list": true,
```
Comment (BirjuVachhani) [inline lib/src/data/json_config.dart:23]: default value should be `false` as not everybody will be using this.

File: lib/src/data/yaml_config.dart [hunk 2]
```diff
@@ -29,6 +29,9 @@ export: true
 # This allows you to import all the generated references with 1 single import!
 use_part_of: true
 
+# Generates a variable that contains a list of all asset values.
+use_references_list: true
```
Comment (BirjuVachhani) [inline lib/src/data/yaml_config.dart:33]: default value should be `false` as not everybody will be using this.

**Review by BirjuVachhani — CHANGES_REQUESTED** [review]

**Comment by BirjuVachhani** [comment]
@zemcov-surf Nice work!! Just minor changes and it is good to be merged.

File: lib/src/data/yaml_config.dart (hunk 2 above)
Comment (zemcov-surf) [reply lib/src/data/yaml_config.dart:33]: Good point)

File: lib/src/data/json_config.dart (hunk 1 above)
Comment (zemcov-surf) [reply lib/src/data/json_config.dart]: Changed it

**Comment by BirjuVachhani** [comment]
Merged. @zemcov-surf Thanks for your contribution! Will release this soon!

**Pull request merged**